File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number}
     */
    subarraySum(nums: number[], k: number): number {
        const prefix: number = 0
        const myMap = new Map<number, number>; 
        let res: number = 0
        for (const n of nums) {
            const diff: number = prefix - n
            if (myMap.has(diff)) {
                res += myMap.get(diff)
            }
            myMap.set(prefix, (myMap.get(prefix) ?? 0) + 1)
        }

        return res
    }
}

File: Data Structures & Algorithms/subsets/submission-0.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        def dfs(nums, temp):
            if not nums:
                res.append(temp)
                return
            temp.append(nums[0])
            dfs(nums[1:], temp)
            temp.pop()
            dfs(nums[1:], temp)
        return res